source/CMakeLists.txt

```diff
@@ -24,6 +24,84 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+set(GRAMMAR_PROCESSING_SCRIPT ${spirv-tools_SOURCE_DIR}/utils/generate_grammar_tables.py)
+
+# macro() definitions are used in the following because we need to append .inc
+# file paths into some global lists (*_CPP_DEPENDS). And those global lists are
+# later used by set_source_files_properties() calls.
+# function() definitions are not suitable because they create new scopes.
+macro(spvtools_core_tables VERSION)
+  set(GRAMMAR_JSON_FILE ${CMAKE_CURRENT_SOURCE_DIR}/spirv-${VERSION}.core.grammar.json)
+  set(GRAMMAR_INSTS_INC_FILE ${spirv-tools_BINARY_DIR}/core.insts-${VERSION}.inc)
+  set(GRAMMAR_KINDS_INC_FILE ${spirv-tools_BINARY_DIR}/operand.kinds-${VERSION}.inc)
+  add_custom_command(OUTPUT ${GRAMMAR_INSTS_INC_FILE} ${GRAMMAR_KINDS_INC_FILE}
+    COMMAND ${PYTHON_EXECUTABLE} ${GRAMMAR_PROCESSING_SCRIPT}
+      --spirv-core-grammar=${GRAMMAR_JSON_FILE}
+      --core-insts-output=${GRAMMAR_INSTS_INC_FILE}
+      --operand-kinds-output=${GRAMMAR_KINDS_INC_FILE}
+    DEPENDS ${GRAMMAR_PROCESSING_SCRIPT} ${GRAMMAR_JSON_FILE}
+    COMMENT "Generate info tables for SPIR-V v${VERSION} core instructions and operands.")
+  list(APPEND OPCODE_CPP_DEPENDS "${GRAMMAR_INSTS_INC_FILE}")
```

Comment:
Perhaps it would be better to put the quotes in the `GRAMMAR_INSTS_INC_FILE` definition.  When passing an argument, the CMake convention is to put quotes around something if we want to treat it as a list.